### ISSUE TITLE
Continue Downstream Testing

### ISSUE THREAD

**Issue by craig8** [VOLTTRON/volttron-core#11]
Currently the downstream testing doesn't use a new artifact from a build.  The workflow for a new PR should be as follows:

![image](https://user-images.githubusercontent.com/3979063/153257475-8f99837d-c8b1-4897-bab7-bb1e0d0f90c9.png)

Image created from sequencediagrams.org using

````
title Downstream Workflow

User->volttronpr: Submits pull request
volttronpr->volttron-dev.whl: Creates dev wheel
volttronpr->downstream.test: Executes Downstream Tests
downstream.test->volttron-dev.whl: Uses dev wheel
downstream.test->volttronpr: Pass with new dev wheel
volttronpr->volttronadmin: Admin merges pr
````

**Comment by bonicim** [comment]
The design suggests that we need the capability to share artifacts across Workflows across different repos. In our specific use-case, we need the capability for downstream tests to receive or get the wheels built by volttron-core.

**Comment by craig8** [comment]
Correct.

**Comment by bonicim** [comment]
This looks like a promising solution to do that: https://github.com/marketplace/actions/download-workflow-artifact

**Comment by bonicim** [comment]
To confirm what exactly is happening when the downstream testing gets the the wheel from volttron-core, the following steps are what I think will occur in the Workflows.

The upstream workflow: 

* Build the wheel
* Store it somewhere for later reuse from the downstream testing workflow
* Send dispatch to downstream testing workflow

Downstream testing workflow:

* Receives dispatch from upstream workflow
* Get the wheel created by the upstream workflow
* Install that wheel using `pip install dist/volttron-0.1.0-py3-none-any.whl`
* Start volttron in the background `python -m volttron.server -vv -l volttron.log &`
* Install the agent to be tested. Example: `python -m volttron.commands.control -vv install volttron_openadr_ven --tag openadr --agent-config config_local.json`
* Check the status: `python -m volttron.commands.control -vv status`
* Run any tests on the installed agent

**Comment by craig8** [comment]
Only after 7 the report of whether 1-7 should be reported back to upstream 3 somehow.

**Comment by bonicim** [comment]
Reopening since more steps need to be implemented.
